migration/v0.16.x-v0.17.x/migrate-dex-additional-static-clients.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -euo pipefail
+
+: "${CK8S_CONFIG_PATH:?Missing CK8S_CONFIG_PATH}"
+
+here="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
+
+secret="${CK8S_CONFIG_PATH}/secrets.yaml"
+secret_tmp="${CK8S_CONFIG_PATH}/secrets_tmp.yaml"
+sops_conf="${CK8S_CONFIG_PATH}/.sops.yaml"
+
+update_script=$(mktemp -p /tmp update-script.XXXXXXXXXX.yaml)
+cp "${here}/dex-update-script.yaml" "${update_script}"
+
+# read all dex's static clients from ck8s cluster
+static_clients_yaml=$(./bin/ck8s ops kubectl sc get secret dex --namespace dex --output json | jq --raw-output '.data."config.yaml"' | base64 --decode | yq read - staticClients)
+
+# delete clients that are part of a standard v0.16.x installation so only custom clients remain
+static_clients_yaml=$(echo "$static_clients_yaml" | \
+    yq delete - "id==kubelogin" | \
+    yq delete - "id==grafana" | \
+    yq delete - "id==grafana-ops" | \
+    yq delete - "id==harbor" | \
+    yq delete - "id==kibana-sso" | \
+    yq delete - "id==grafana")
+
+# merge custom clients from ck8s cluster into secrets config under additionalStaticClients
+echo "$static_clients_yaml" | yq prefix - "[0].value" | yq merge --inplace "${update_script}" -
+sops -d "${secret}" | \
+    yq write --script "${update_script}" - | \
+    sops --config "${sops_conf}" --input-type=yaml --output-type=yaml -e /dev/stdin > "${secret_tmp}"
+
+cp "${secret}" "${secret}.bak"
```

Comment:
Just added this as a precaution, if this is something we want to keep in the script I can add a instruction to the migration instructions to delete it after verification?

Reply:
I'm not sure, ideally everyone should have their config path version controlled so then this is not needed. But it's always nice to be able to have a backup.

Maybe add a note in the migration guide after the script step with something like, `verify that the script added all your custom static clients and delete the backup`